pkg/addon/push.go

```diff
@@ -0,0 +1,296 @@
+		url = strings.Replace(repo.Config.URL, "cm://", "https://", 1)
+	}
+
+	cmClient, err := cm.NewClient(
+		cm.URL(url),
+		cm.Username(username),
+		cm.Password(password),
+		cm.AccessToken(p.AccessToken),
+		cm.AuthHeader(p.AuthHeader),
+		cm.ContextPath(p.ContextPath),
+		cm.CAFile(p.CaFile),
+		cm.CertFile(p.CertFile),
+		cm.KeyFile(p.KeyFile),
+		cm.InsecureSkipVerify(p.InsecureSkipVerify),
+		cm.Timeout(p.Timeout),
+	)
+
+	if err != nil {
+		return err
+	}
+
+	// Use a temporary dir to hold packaged .tgz Charts
+	tmp, err := ioutil.TempDir("", "helm-push-")
+	if err != nil {
+		return err
+	}
+	defer func(path string) {
+		_ = os.RemoveAll(path)
+	}(tmp)
+
+	// Package Chart into .tgz packages for uploading to ChartMuseum
+	chartPackagePath, err := cmhelm.CreateChartPackage(chart, tmp)
+	if err != nil {
+		return err
+	}
+
+	_, _ = fmt.Fprintf(os.Stderr, "Pushing %s to %s... ",
+		color.New(color.Bold).Sprintf(filepath.Base(chartPackagePath)),
+		formatRepoNameAndURL(p.RepoName, repo.Config.URL),
+	)
+
+	// Push Chart to ChartMuseum
+	resp, err := cmClient.UploadChartPackage(chartPackagePath, p.ForceUpload)
+	if err != nil {
+		return err
+	}
+
+	return handlePushResponse(resp)
+}
+
+// GetHelmRepo searches for a Helm repo by name.
+// By saying name, it can actually be a URL or a name.
+// If a URL is provided, a temp repo object is returned.
+// If a name is provided, we will try to find it in local addon registries (only Helm type).
+func GetHelmRepo(ctx context.Context, c client.Client, repoName string) (*cmhelm.Repo, error) {
+	var repo *cmhelm.Repo
+	var err error
+
+	// If RepoName looks like a URL (https / http), just create a temp repo object.
+	// We do not look for it in local addon registries.
+	if regexp.MustCompile(`^https?://`).MatchString(repoName) {
+		repo, err = cmhelm.TempRepoFromURL(repoName)
+		if err != nil {
+			return nil, err
+		}
+		return repo, nil
+	}
+
+	// Otherwise, search for in it in the local addon registries.
+	ds := NewRegistryDataStore(c)
+	registries, err := ds.ListRegistries(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	var matchedEntry *helmrepo.Entry
+
+	// Search for the target repo name in addon registries
+	for _, reg := range registries {
+		// We are only interested in Helm registries.
+		if reg.Helm == nil {
+			continue
+		}
+
+		if reg.Name == repoName {
+			matchedEntry = &helmrepo.Entry{
+				Name:     reg.Name,
+				URL:      reg.Helm.URL,
+				Username: reg.Helm.Username,
+				Password: reg.Helm.Password,
+			}
+			break
+		}
+	}
+
+	if matchedEntry == nil {
+		return nil, fmt.Errorf("we cannot find Helm repository %s. Make sure you hava added it using `vela addon registry add` and it is a Helm repository", repoName)
+	}
+
+	// Use the repo found locally.
+	repo = &cmhelm.Repo{ChartRepository: &helmrepo.ChartRepository{Config: matchedEntry}}
+
+	return repo, nil
+}
+
+// SetFieldsFromEnv sets fields in PushCmd from environment variables
+func (p *PushCmd) SetFieldsFromEnv() {
```

Comment:
do these env variable align with the HELM official site?

Reply:
Yes, they are kept the same.

Reply:
great